### PR TITLE
Update Custom Race Presets for StartTime

### PR DESCRIPTION
**Pull request by mathuaerknedam**
Didn't change variable names (Torn didn't), just made it work with Torn's logical change from delay to start time.

### CODE REVIEW

**Comment by Manuito83** [comment]
Thanks! Is this something we were having issues with? I guess that the script version would need to be updated as well.

File: userscripts/Custom Race Presets (Torn PDA).js [hunk 1]
```diff
@@ -33,15 +33,15 @@ var presets = [{
...
+	waitTime: "ASAP",
 	password: "",
 },
 {
 	name: "1hr Start - Docks",
 	maxDrivers: 100,
 	trackName: "Docks",
 	numberOfLaps: 100,
-	waitTime: 60,
+	waitTime: "Today 23:45 TCT",
```
Comment (Kwack-Kwack) [inline userscripts/Custom Race Presets (Torn PDA).js:44]: We can't just hardcode a random time here. Leave the waitTime as the number of minutes, and we'll handle it down below.

File: userscripts/Custom Race Presets (Torn PDA).js [hunk 2]
```diff
@@ -77,6 +76,10 @@ function fillPreset(index) {
 		$('#select-allow-upgrades').selectmenu();
 		$('#select-allow-upgrades-menu > li:contains(' + race.upgradesAllowedString + ')').mouseup();
 	}
+	if ("waitTime" in race) {
+		$('#wait-time').selectmenu();
+		$('#wait-time-menu > li:contains(' + race.waitTime + ')').mouseup();
```
Comment (Kwack-Kwack) [inline userscripts/Custom Race Presets (Torn PDA).js:81]: if `race.waitTime` > 1 we should get the utc time string in x minutes, and round it up to the nearest 15 minutes. Will need to handle the dates (check if it's today, tomorrow, or the date beyond that).
If it's not greater than 1, set it to ASAP

**Review by Kwack-Kwack — CHANGES_REQUESTED** [review]
Thanks for catching this, I didn't even notice they changed this. I've left some comments, but to sum it up we should recode the logic rather than the template (which will break both user's templates & the current behaviour of the script).

To be frank the script will need a proper rewrite at some point for a couple issues, but this should work fine as a hotfix. 

Also note that you'll need to increment the version number after making these changes as well.

File: userscripts/Custom Race Presets (Torn PDA).js (hunk 2 above)
Comment (Kwack-Kwack) [reply userscripts/Custom Race Presets (Torn PDA).js:81]: I'm not much of a racer myself so you're quite possibly right, I'm just avoidant of breaking existing behaviours and presets in the process.

Perhaps we could do both? Handle strings as precise times, and numbers as "in x minutes"? That way existing presets should continue working as expected, and people stewarding can also set specific times.

Comment (mathuaerknedam) [reply userscripts/Custom Race Presets (Torn PDA).js:81]: Sure, I'll see what I can do. Though the preset time have been broken for over two years, so I'm not sure anyone will notice. https://www.torn.com/forums.php#/p=threads&f=21&t=16283415&b=0&a=0

**Comment by Manuito83** [comment]
We still need to work on this, correct?

**Comment by Kwack-Kwack** [comment]
> We still need to work on this, correct?

Coincidentally I started rewriting this two days ago 😅 still ironing out some minor bugs here and there. I'll close this PR and open my own soon™️

**Pull request closed without merging**